packages/calypso-build/bin/transpile.js

```diff
@@ -32,18 +32,40 @@ for ( const arg of process.argv.slice( 2 ) ) {
 // root directory (set as cwd) which isn't an ancestor of any of the source files.
 const testIgnorePattern = path.join( dir, '**/test/**' );
 
-console.log( 'Building %s', dir );
+// Try a couple of levels up as well to get it working in a monorepo where deps are hoisted.
+const possibleRoots = [ root, path.dirname( root ), path.dirname( path.dirname( root ) ) ];
+let thisRootActuallyWorks = root;
+for ( const testRoot of possibleRoots ) {
+	thisRootActuallyWorks = testRoot;
+	try {
+		// The -f flag essentially gets babel to exit with a 0 exit code. Otherwise,
+		// it fails because the required arguments don't exist. This makes it easier
+		// to test that the binary at least exists.
+		execSync( 'npx --no-install babel -f this-is-fake-and-doesnt-exist', {
```

Comment:
I'm not very happy with this solution, but `yarn run -T babel` does not work, `yarn exec babel` does not work, etc. And importing babel doesn't leave us with a clear JS API for doing the exact same thing we're trying to do here.

Reply:
I'm also not happy about using `npx` here, because the `npx` API can change in future versions. (In fact, --no-install is already deprecated in newer versions.) But we don't control the npx version from package.json. It'd be a lot better to rely on something inside the module tree, IMO

Reply:
I changed this to use `require.resolve` and access @babel/cli directly.